*: improve some tests

knz opened the pull request:
This PR contains the test improvements prerequisite to #110001.
Epic: CRDB-18499

knz committed:
serverccl: opt out a test from virtual clusters

Referring to followup issue #110002.

Release note: None

knz committed:
sqlccl: fix a test

Release note: None

knz committed:
streamingest: fix a test

Release note: None

knz committed:
clouccl/azure: simplify test

Release note: None

knz committed:
cloudccl/amazon: simplify tests

Release note: None

knz committed:
cloudccl/gcp: simplify tests

Release note: None

knz committed:
cloudccl/externalconn: fix and simplify test code

Release note: None

knz committed:
cli: fix a test

Release note: None

knz committed:
catalog/descs: fix a test

Release note: None

knz committed:
server: properly propagate `TestingInsecureWebAccess`

Release note: None

knz committed:
security: improve TestRotateCerts

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110008)
<!-- Reviewable:end -->

knz committed:
spanconfig: various test improvements

Release note: None

knz committed:
sql/copy: fix some tests

Release note: None

knz committed:
serverutils: silence a warning

Release note: None

knz committed:
server/authserver: fix a test

Release note: None

knz committed:
application_api: link a test to a followup issue

Release note: None

knz committed:
kvstreamer: fix some tests

Release note: None

knz committed:
testutils: lift `NodeDialer` to `ApplicationLayerInterface`

The dialer for a given SQL layer dials across SQL servers using
instance IDs.

Release note: None

knz committed:
tracing/collector: silence some warnings

Release note: None

knz committed:
kvserver/batcheval: fix a test and simplify test code

Release note: None

knz committed:
sql/gcjob: opt a test out of cluster virtualization

and link it to follow-up issue #110014.

Release note: None

knz committed:
protectedts/ptreconcile: fix and simplify test

Release note: None

knz committed:
ts: fix some tests and silence some warnings

Release note: None

knz committed:
kvprober: opt tests out of cluster virtualization

Release note: None

knz committed:
server: pick up heap/goroutine dump dirs in sec tenant servers

This incomplete configuration was noticed by `TestStatusGetFiles` once
enabled to run over secondary tenants. This extra test coverage is
enabled in the subsequent commit.

Release note: None

knz committed:
server/storage_api: fix tests and simplify test code

Release note: None

yuzefovich commented on pkg/sql/copy/copy_test.go:
```diff
@@ -88,17 +88,20 @@ const csvData = `%d|155190|7706|1|17|21168.23|0.04|0.02|N|O|1996-03-13|1996-02-1
+					defer log.Scope(t).Close(t)
+
+					srv := serverutils.StartServerOnly(t, base.TestServerArgs{
 						Settings: cluster.MakeTestingClusterSettings(),
```
nit: no need to create a settings container here.

yuzefovich commented on pkg/sql/copy/copy_test.go:
```diff
@@ -243,19 +246,21 @@ func expandErrorString(err error) string {
-		s := serverutils.StartServerOnly(t, base.TestServerArgs{
+		defer log.Scope(t).Close(t)
+		srv := serverutils.StartServerOnly(t, base.TestServerArgs{
 			Settings: cluster.MakeTestingClusterSettings(),
```
nit: ditto

yuzefovich commented on pkg/server/authserver/authentication_test.go:
```diff
@@ -119,7 +133,7 @@ func TestSSLEnforcement(t *testing.T) {
 	insecureContext := newRPCContext(true, username.TestUserName())
 
 	kvGet := &kvpb.GetRequest{}
-	kvGet.Key = roachpb.Key("/")
```
nit: this seems like not an equivalent change - why are we not appending `"/"` to the tenant prefix?

yuzefovich commented on pkg/kv/kvserver/batcheval/cmd_export_test.go:
```diff
@@ -436,42 +435,38 @@ func TestExportRequestWithCPULimitResumeSpans(t *testing.T) {
 
 	ctx := context.Background()
 	rng, _ := randutil.NewTestRand()
-	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{
```
It'd be good to get @stevendanna to sign off on this test going from multi-node cluster to single server. If it's acceptable, then we should remove the SCATTER command as well as the scatter setting override since it's meaningless in single node setup.

stevendanna replied:
I think we need to keep this as a multi-node test. Part of what this is testing is that the DistSender is able to correctly decide whether or not a request that needs to be split across multiple ranges can be parallelised and combined or not.

yuzefovich commented on pkg/kv/kvserver/batcheval/cmd_is_span_empty_test.go:
```diff
@@ -95,7 +102,13 @@ func TestIsSpanEmpty(t *testing.T) {
+	splitRangeOrFatal(mkKey("b"))
+	expectedCalls := int64(4)
+	if srv.TenantController().StartedDefaultTestTenant() {
+		// TODO(kv): investigate why there is one more call to IsSpanEmpty
```
nit: this seems like a TODO worthy of an explicit issue.

yuzefovich commented on pkg/kv/kvprober/kvprober_integration_test.go:
```diff
@@ -378,6 +378,9 @@ func initTestServer(
 	t *testing.T, knobs base.TestingKnobs,
 ) (serverutils.TestServerInterface, *gosql.DB, *kvprober.Prober, func()) {
 	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		// KV probes always to to the storage layer.
```
nit: `s/to to/go to/`.

yuzefovich reviewed (commented):
I have some nits.

stevendanna commented on pkg/server/server_sql.go:
```diff
@@ -1373,6 +1374,7 @@ func newSQLServer(ctx context.Context, cfg sqlServerArgs) (*SQLServer, error) {
 		sessionRegistry:                cfg.sessionRegistry,
 		closedSessionCache:             cfg.closedSessionCache,
 		jobRegistry:                    jobRegistry,
+		nodeDialer:                     cfg.podNodeDialer,
```
I wonder if we want a better name here since in the context of a MT sql server we have two dialers and `nodeDialer` doesn't make it immediately clear which is which. We could perhaps name this one SQLInstanceDialer.  

I don't mind if we want to punt this to a follow-up.

stevendanna commented on pkg/server/authserver/authentication_test.go:
```diff
@@ -89,20 +90,33 @@ func TestSSLEnforcement(t *testing.T) {
+			`ALTER TENANT [$1] GRANT CAPABILITY can_view_node_info=true`, serverutils.TestTenantID().ToUint64())
+		require.NoError(t, err)
+
+		serverutils.WaitForTenantCapabilities(t, srv, serverutils.TestTenantID(), map[tenantcapabilities.ID]string{
+			tenantcapabilities.CanViewNodeInfo: "true",
+		}, "")
```
Not necessary for this PR, but I wonder what you think of integrating a call to `ts.tenantCapabilitiesWatcher.TestingRestart()` into WaitForTenantCapabilities to avoid these tests slowing down too much when we add these wait calls.

knz replied:
Ok. Added a commit at the end to do that.

stevendanna reviewed (commented):
I didn't add much on top of Yahor's review. But there is one required change in my opinion.

Huge thanks for continuing to drive this forward.

knz committed:
colflow: avoid a warning

Release note: None

knz committed:
serverutils: lift `WaitForTenantCapabilities` into TestServer

This way the optimizing call to `TestingRestart` on the cap watcher is
performed in all tests that use this API.

Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r2, 13 of 13 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110008#-:-NdpkJCS49_vw0t1mowA:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier, @herkolategan, @michae2, @srosenberg, and @stevendanna)



<!-- Sent from Reviewable.io -->

abarganier commented on pkg/ccl/serverccl/server_controller_test.go:
```diff
@@ -535,8 +535,11 @@ func TestServerControllerLoginLogout(t *testing.T) {
-	s := serverutils.StartServerOnly(t, base.TestServerArgs{})
-	defer s.Stopper().Stop(ctx)
+	srv := serverutils.StartServerOnly(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(110002),
```
> base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(110002)

☠️ 😂

abarganier commented on pkg/server/server_controller_new_server.go:
```diff
@@ -246,6 +246,8 @@ func makeSharedProcessTenantServerConfig(
 	baseCfg.SpanConfigsDisabled = kvServerCfg.BaseConfig.SpanConfigsDisabled
 	baseCfg.EnableDemoLoginEndpoint = kvServerCfg.BaseConfig.EnableDemoLoginEndpoint
 	baseCfg.DefaultZoneConfig = kvServerCfg.BaseConfig.DefaultZoneConfig
+	baseCfg.HeapProfileDirName = kvServerCfg.BaseConfig.HeapProfileDirName
```
nit: does this impact tests? Or is it just some cleanup along the way.

knz replied:
This is exercised in tests; its lack was noticed by a failure in #110001.

abarganier reviewed (approved):
LGTM, I appreciate the clarity that's added from this overhaul.

knz commented:
TFYRs!

bors r=stevendanna,yuzefovich,abargainier
